go-controller/pkg/testing/exec.go

```diff
@@ -20,10 +20,12 @@ type FakeExec struct {
 	// We will in such a case ignore order when comparing all executed commands during the run of a test case.
 	// This is important when defining test cases with multiple resources (or multiple resource watchers) of
 	// the same type and not being able to rely on a deterministic order of incomming watch events.
-	looseCompare     bool
-	expectedCommands []*ExpectedCmd
-	executedCommands []string
-	mu               sync.Mutex
+	looseCompare       bool
+	expectedCommands   []*ExpectedCmd
+	executedCommands   []string
+	mu                 sync.Mutex
+	receivedUnexpected bool
```

Comment:
Nit, we could just use the string variable: `unexpectedCommand` for this change, since it will always be non-empty when `receivedUnexpected` is true.

Reply:
Yep I don't have a strong opinion here (and performances don't def matter here). I think having a bool is slightly better in terms of readability